src/fprime_gds/common/loaders/constant_json_loader.py

```diff
@@ -0,0 +1,52 @@
+"""
+constant_json_loader.py:
+
+Loads flight dictionary (JSON) and returns dictionaries of all constants
+defined in the dictionary.
+
+@author thomas-bc
+"""
+
+from fprime_gds.common.loaders.json_loader import JsonLoader
+from fprime_gds.common.data_types.exceptions import GdsDictionaryParsingException
+
+
+class ConstantJsonLoader(JsonLoader):
+    """Class to load Python objects representing constants from the JSON dictionary"""
+
+    CONSTANTS_FIELD = "constants"
+
+    def construct_dicts(self, _):
+        """
+        Constructs and returns python dictionaries keyed on id and name
+
+        Args:
+            _: Unused argument (inherited)
+        Returns:
+            A tuple with two Fw type dictionaries (python type dict):
+            (id_dict, name_dict). The keys should be the type id and
+            name fields respectively and the values should be type name
+            strings. Note: An empty id dictionary is returned since there
+            are no id fields in the Fw type alias JSON dictionary entries.
+        """
+        id_dict = {}
+        name_dict = {}
+
+        if self.CONSTANTS_FIELD not in self.json_dict:
+            raise GdsDictionaryParsingException(
```

Comment:
Does this make it dictionary parsing incompatible with older JSON dictionaries?

Reply:
Ah I thought `constants` have always had their field in the dictionary, but seems they've been added later on. Will fix.